tiledb/sm/filesystem/vfs.cc

```diff
@@ -1210,39 +1211,94 @@ Status VFS::read_impl(
 #else
     return posix_.read(uri.to_path(), offset, buffer, nbytes);
 #endif
-  }
-  if (uri.is_hdfs()) {
+  } else if (uri.is_hdfs()) {
 #ifdef HAVE_HDFS
     return hdfs_->read(uri, offset, buffer, nbytes);
 #else
     return LOG_STATUS(
         Status::VFSError("TileDB was built without HDFS support"));
 #endif
   }
+
+  // Avoid a read if the requested buffer can be read from the
+  // read cache. Note that we intentionally do not use a read
+  // cache for local files because we rely on the operating
+  // system's file system to cache readahead data in memory.
+  // Additionally, we do not perform readahead with HDFS.
+  if (read_ahead_cache_.read(uri, offset, buffer, nbytes)) {
```

Comment:
I think this should be moved after the `do_read_ahead` check and read from the cache only if a read ahead is required. This will save a mutex lock inside the LRU cache object when reading.

Reply:
Obsolete